src/ORM/DataObject.php

```diff
@@ -459,8 +459,20 @@ protected function duplicateManyManyRelation($sourceObject, $destinationObject,
         // Copy all components from source to destination
         $source = $sourceObject->getManyManyComponents($manyManyName);
         $dest = $destinationObject->getManyManyComponents($manyManyName);
+
+        if ($source instanceof UnsavedRelationList) {
```

Comment:
I'd consider swapping this round so 

```php
if ($source instanceof ManyManyList) {
    $extraFieldNames = $source->getExtraFields();
} else { 
    $source = []; 
}
```

That way we are using the API we know exists on `ManyManyList` rather than *not* using it on  one we know doesn't...

Reply:
good point... commit incoming